src/xunit.runner.visualstudio/Sinks/VsDiscoverySink.cs

```diff
@@ -190,11 +190,11 @@ void HandleCancellation(MessageHandlerArgs args)
         void HandleTestCaseDiscoveryMessage(MessageHandlerArgs<ITestCaseDiscoveryMessage> args)
         {
             var testCase = args.Message.TestCase;
-            var testMethod = $"{testCase.TestMethod.TestClass.Class.Name}.{testCase.TestMethod.Method.Name}";
-            if (lastTestMethod != testMethod)
+            var testClass = $"{testCase.TestMethod.TestClass.Class.Name}";
+            if (lastTestClass != testClass)
```

Comment:
Optionally, we could also include a max count, so max batches of 500? (or some other arbitrary number)